### PR TITLE
Elasticsearch: Remove browser access mode

### LINKED ISSUE #12757: Remove Elasticsearch direct/browser access mode
Why?
* Very high maintenance burden to have multiple implementations in frontend and backend (elasticsearch is Grafana's most advanced datasource since it has a very complicated request/response model)
* Very confusing for end user that graph panel render based on frontend implementation and alerting evaluates data based on backend implementation
* To be able to use alerting there's no support for direct/browser access mode
* Security - No way for end user to access an index not configured in datasource
* Supporting both frontend and backend slows down the development of adding new features to  datasource

Would appreciate feedback for those using direct/browser access mode and why they're needing that.

### PR DESCRIPTION
**Pull request by gabor**
Fixes https://github.com/grafana/grafana/issues/12757

in the Elasticsearch datasource configuration, you can choose the access mode, `browser` or `server`. the `browser` mode was deprecated in grafana7.4.0, and now it has been removed.

to be the least surprising and confusing, i chose the solution where we do not change anything automatically, except browser-mode stops working. in such a situation we explain to the user why it is not working and how to fix the problem. the user has to change the configuration.

in the documentation i removed all the explanations for this setting because now there are only two possible settings: works & does_not_work.

how to test:
- setup an elasticsearch datasource.
- switch the access-mode to `browser`
- you should see an error message at the top
- press [save&test], you should get the same error-message
- now go to explore-mode, and try to query this datasource
- you should get the same error-message again
- now go back to datasource-config
- switch the access-mode to `server`
- the error-message should vanish
- press [save&test], you should get no error-message
- go to explore, run a query, you should get no error-message

# Release notice breaking change

In Elasticsearch, browser access mode was deprecated in grafana 7.4.0 and removed in 9.0.0. If you used this mode, please switch to server access mode on the datasource configuration page.

### CODE REVIEW

File: docs/sources/datasources/elasticsearch.md [hunk 1]
```diff
@@ -25,32 +25,12 @@ Supported Elasticsearch versions:
...
-If you select Browser access you must update your Elasticsearch configuration to allow other domains to access
-Elasticsearch from the browser. You do this by specifying these two options in your **elasticsearch.yml** config file.
-
-```bash
-http.cors.enabled: true
-http.cors.allow-origin: "*"
-```
+| Name      | Description                                                                                                           |
+| --------- | --------------------------------------------------------------------------------------------------------------------- |
+| `Name`    | The data source name. This is how you refer to the data source in panels and queries.                                 |
```
Comment (brendamuir) [inline docs/sources/datasources/elasticsearch.md:30]:
```suggestion
| `Name`    | Data source name. This is how you refer to the data source in panels and queries.                                 |
```

File: docs/sources/datasources/elasticsearch.md [hunk 2]
```diff
@@ -25,32 +25,12 @@ Supported Elasticsearch versions:
...
+| `Default` | Default data source means that it will be pre-selected for new panels.                                                |
```
Comment (brendamuir) [inline docs/sources/datasources/elasticsearch.md:31]:
```suggestion
| `Default` | Data source is pre-selected for new panels.                                                |
```

File: docs/sources/datasources/elasticsearch.md [hunk 3]
```diff
@@ -25,32 +25,12 @@ Supported Elasticsearch versions:
...
+| `Url`     | The HTTP protocol, IP, and port of your Elasticsearch server.                                                         |
+| `Access`  | This setting is only here for historical reasons. Use "Server (default)", otherwise the datasource will not function. |
```
Comment (brendamuir) [inline docs/sources/datasources/elasticsearch.md:33]:
```suggestion
| `Access`  | Do not use Access. Use "Server (default)" or the datasource won't function. |
```

File: public/app/plugins/datasource/elasticsearch/configuration/ConfigEditor.tsx [hunk 4]
```diff
@@ -32,8 +32,9 @@ export const ConfigEditor = (props: Props) => {
   return (
     <>
       {options.access === 'direct' && (
-        <Alert title="Deprecation Notice" severity="warning">
-          Browser access mode in the Elasticsearch datasource is deprecated and will be removed in a future release.
+        <Alert title="Error" severity="error">
+          Browser access mode in the Elasticsearch datasource is not available anymore. Please switch to server access
```
Comment (brendamuir) [inline public/app/plugins/datasource/elasticsearch/configuration/ConfigEditor.tsx:36]:
```suggestion
          Browser access mode in the Elasticsearch datasource is no longer available. Switch to server access
```

File: public/app/plugins/datasource/elasticsearch/datasource.ts [hunk 5]
```diff
@@ -132,6 +134,13 @@ export class ElasticDatasource
     data?: undefined,
     headers?: BackendSrvRequest['headers']
   ): Observable<any> {
+    if (!this.isProxyAccess) {
+      const error = new Error(
+        'Browser access mode in the Elasticsearch datasource is not available anymore. Please switch to server access mode.'
```
Comment (brendamuir) [inline public/app/plugins/datasource/elasticsearch/datasource.ts:139]:
```suggestion
        'Browser access mode in the Elasticsearch datasource is no longer available. Switch to server access mode.'
```

**Review by brendamuir — APPROVED** [review]

**Review by joey-grafana — APPROVED** [review]
It works, although I can't test if an error message is displayed when I switch the access-mode to `browser` as when I press the Test I see the error but it does not actually save the datasource.

Maybe we should tell the user that they can't actually save the mode to browser?

Screenshot 2022-05-17 at 19 55 04

I just removed the `!` in the elastic datasource.ts to _flip the switch_ and test the logic.

Screenshot 2022-05-17 at 19 56 57

**Review by brendamuir — APPROVED** [review]

**Pull request merged**